src/portal-impl.c

```diff
@@ -387,61 +387,114 @@ load_portal_configuration_for_dir (gboolean    opt_verbose,
   return NULL;
 }
 
+/*
+ * Returns: %TRUE if configuration was found in @dir
+ */
+static gboolean
+load_config_directory (const char *dir,
+                       const char **desktops,
+                       gboolean opt_verbose)
+{
+  g_autoptr(PortalConfig) conf = NULL;
+
+  for (size_t i = 0; desktops[i] != NULL; i++)
+    {
+      g_autofree char *portals_conf = g_strdup_printf ("%s-portals.conf", desktops[i]);
+
+      conf = load_portal_configuration_for_dir (opt_verbose, dir, portals_conf);
+
+      if (conf != NULL)
+        {
+          if (opt_verbose)
+            g_debug ("Using portal configuration file '%s/%s' for desktop '%s'",
+                     dir, portals_conf, desktops[i]);
+
+          config = g_steal_pointer (&conf);
+          return TRUE;
+        }
+    }
+
+  conf = load_portal_configuration_for_dir (opt_verbose, dir, "portals.conf");
+
+  if (conf != NULL)
+    {
+      if (opt_verbose)
+        g_debug ("Using portal configuration file '%s/%s' for non-specific desktop",
+                 dir, "portals.conf");
+
+      config = g_steal_pointer (&conf);
+      return TRUE;
+    }
+
+  return FALSE;
+}
+
+#define XDP_SUBDIR "xdg-desktop-portal"
+
 void
 load_portal_configuration (gboolean opt_verbose)
 {
-  g_autoptr(PortalConfig) conf = NULL;
   g_autofree char *user_portal_dir = NULL;
+  const char * const *dirs;
+  const char * const *iter;
   const char **desktops;
   const char *portal_dir;
 
+  desktops = get_current_lowercase_desktops ();
+
   /* We need to override this in the tests */
   portal_dir = g_getenv ("XDG_DESKTOP_PORTAL_DIR");
```

Comment:
I left this as an intentionally undocumented implementation detail, since it's only there for the tests as far as I can see.